src/BenchmarkDotNet/Validators/ExecutionValidatorBase.cs

```diff
@@ -44,7 +44,14 @@ public IEnumerable<ValidationError> Validate(ValidationParameters validationPara
                     continue;
                 }
 
+                int errorCount = errors.Count;
+
                 ExecuteBenchmarks(benchmarkTypeInstance, typeGroup, errors);
+
+                if (errorCount == errors.Count)
```

Comment:
I think that we should try to execute the cleanup no matter if there were any exceptions thrown when executing the benchmark.

Example:
Setup - open db connection
Execute - single sql query
Cleanup - close the db connection (mandatory)

Reply:
Good point, will make the change.